lab/src/Microsoft.ServiceModel.Syndication/src/Atom10FeedFormatter.cs

```diff
@@ -691,35 +798,33 @@ private DateTimeOffset DateFromString(string dateTimeString, XmlReader reader)
             {
                 return utcTime;
             }
-            throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(
-                new XmlException(FeedUtils.AddLineInfo(reader,
-                SR.ErrorParsingDateTime)));
 
-            throw new NullReferenceException(); // <----- has to be fixed
+            throw new XmlException(FeedUtils.AddLineInfo(reader, SR.ErrorParsingDateTime));
         }
-
-        private void ReadCategory(XmlReader reader, SyndicationCategory category)
+        
+        private async Task ReadCategoryAsync(XmlReaderWrapper reader, SyndicationCategory category)
         {
-            ReadCategory(reader, category, this.Version, this.PreserveAttributeExtensions, this.PreserveElementExtensions, _maxExtensionSize);
+            await ReadCategoryAsync(reader, category, this.Version, this.PreserveAttributeExtensions, this.PreserveElementExtensions, _maxExtensionSize);
         }
 
-        private SyndicationCategory ReadCategoryFrom(XmlReader reader, SyndicationFeed feed)
+        private async Task<SyndicationCategory> ReadCategoryFromAsync(XmlReaderWrapper reader, SyndicationFeed feed)
+
         {
             SyndicationCategory result = CreateCategory(feed);
-            ReadCategory(reader, result);
+            await ReadCategoryAsync(reader, result);
             return result;
```

Comment:
If you modified ReadCategoryAsync to return the passed in SyndicationCategory, you could avoid an await here and simply return the Task<SyndicationCategory> from ReadCategoryAsync